tensorly/metrics/similarity.py

```diff
@@ -0,0 +1,81 @@
+from typing import List
+
+import numpy as np
+from sklearn.preprocessing import normalize
+
+# Authors: Hratch Baghdassarian <hmbaghdassarian@gmail.com>, Erick Armingol <earmingol14@gmail.com>
+# similarity metrics for tensor decompsoitions
+
+def stack_loading_matrices(factors: List[np.ndarray]) -> np.ndarray:
+    """Vertically stack the loading/factor matrices.
+
+    Parameters
+    ----------
+    factors : List[np.ndarray]
+        The loading/factor matrices [A^1 ... A^R] for a decomposed low-rank tensor from its factors.
+
+    Returns
+    -------
+    np.ndarray
+        A vertically stacked matrix of shape [sum(tensor.shape)xR)] 
+    """
+    
+    if not np.all([A.shape[1] for A in factors]):
+        raise ValueError('factors should be a list of loading matrices')
+    return np.vstack(factors)
+
+def calc_corrindex(C: np.ndarray) -> float:
+    """Calculate the CorrIndex.
+
+    Parameters
+    ----------
+    C : np.ndarray
+        absolute value of the product of two MxN matrices that have had their columns L2 normalized, 
+        taking the complex conjugate transpose of the first matrix -- C = |X1^H * X2|
+
+    Returns
+    -------
+    float
+        CorrIndex metric [0,1]; lower score means higher between matrices
+    """
+    
+    MN = C.shape[1]*C.shape[0]
+    return (1/(MN))*(np.sum(np.abs(np.amax(C, 1) - 1)) + np.sum(np.abs(np.amax(C, 0) - 1)))
```

Comment:
To make this work for other backends, you'll want to use the Tensorly operations. You can add `import tensorly as tl` at the top, then most of these can just be changed from `np.X` to `tl.X`. I think `amax` might have to become `max`.